test/SILOptimizer/temp_rvalue_opt_ossa.sil

```diff
@@ -180,6 +185,26 @@ bb0(%0 : $*Builtin.NativeObject):
   %v = tuple ()
   return %v : $()
 }
+
+// CHECK-LABEL: sil [ossa] @dont_allow_copy_take_from_projection
+// CHECK:   [[STK:%[0-9]+]] = alloc_stack
+// CHECK:   copy_addr [take] %1 to [initialization] [[STK]]
+// CHECK: } // end sil function 'dont_allow_copy_take_from_projection'
+sil [ossa] @dont_allow_copy_take_from_projection : $@convention(thin) (@in Two) -> @out Two {
+bb0(%0 : $*Two, %1 : $*Two):
+  %a0 = struct_element_addr %0 : $*Two, #Two.a
+  %b0 = struct_element_addr %0 : $*Two, #Two.b
+  %s = alloc_stack $Two
+  copy_addr [take] %1 to [initialization] %s : $*Two
```

Comment:
Is this an issue in non-ossa code as well?

Reply:
yes. The only difference is that the memory lifetime verifier will not catch it (because it only runs on ossa).

Reply:
The non-ossa version would be completely equivalent and also be handled equivalently in the optimization.

Reply:
Ok.